src/Foundation/DictionaryContainer.cs

```diff
@@ -326,8 +325,12 @@ protected void SetArrayValue (NSString key, NSNumber[]? values)
 
 		protected void SetArrayValue<T> (NSString key, T[]? values)
 		{
-			if (NullCheckAndRemoveKey (key, values == null))
-				Dictionary [key] = NSArray.FromNSObjects (values.Select (x => NSObject.FromObject (x)).ToArray ());
+			if (NullCheckAndRemoveKey (key, values == null)) {
```

Comment:
Looks to me like the problem is that there's no nullability analysis on values == null. `NullCheckAndRemoveKey` does not mark `values` as non-null.

Reply:
I don't think `NullCheckAndRemoveKey` can mark `values` in any way without changing the code in a much bigger way.